crates/primitives-traits/src/receipt.rs

```diff
@@ -28,7 +29,10 @@ pub trait Receipt:
 {
     /// Returns transaction type.
     fn tx_type(&self) -> u8;
+}
 
+/// Extension if [`Receipt`] used in block execution.
+pub trait ReceiptExt: Receipt {
```

Comment:
cool, splitting this makes sense we definitely want this because op can't have this

Reply:
ah true, optimism needs the chain spec too! in general, I'm not convinced that this is the right place for the method...think this approach made the most sense https://github.com/paradigmxyz/reth/pull/11314